examples/timeseries/timeseries_classification_from_scratch.py

```diff
@@ -0,0 +1,237 @@
+    conv1 = keras.layers.BatchNormalization()(conv1)
+    # Finally we add the relu activation function to inject some non-linearity
+    # to the convolutions' output.
+    conv1 = keras.layers.ReLU()(conv1)
+
+    conv2 = keras.layers.Conv1D(filters=64, kernel_size=3, padding="same")(conv1)
+    conv2 = keras.layers.BatchNormalization()(conv2)
+    conv2 = keras.layers.ReLU()(conv2)
+
+    conv3 = keras.layers.Conv1D(filters=64, kernel_size=3, padding="same")(conv2)
+    conv3 = keras.layers.BatchNormalization()(conv3)
+    conv3 = keras.layers.ReLU()(conv3)
+
+    # A global average pooling is used here.
+    # We average the time series over the whole time dimension.
+    # This would reduce drastically the number of parameters,
+    # while enabling the use of the class activation map method for interpretability.
+    # In fact the input time series with m dimensions is averaged
+    # resulting in a vector of m dimensions.
+    gap = keras.layers.GlobalAveragePooling1D()(conv3)
+
+    # The final softmax traditional classifier is used,
+    # with a number of neurons equal to the number of classes.
+    output_layer = keras.layers.Dense(num_classes, activation="softmax")(gap)
+
+    # We link the input and output layer by constructing the keras model.
+    return keras.models.Model(inputs=input_layer, outputs=output_layer)
+
+
+model = make_model(input_shape=x_train.shape[1:])
+keras.utils.plot_model(model, show_shapes=True)
+
+"""
+## Train the model
+
+"""
+
+epochs = 500
+batch_size = 128
+
+callbacks = [
+    keras.callbacks.ModelCheckpoint(
+        "best_model.h5", save_best_only=True, monitor="val_loss"
+    ),
+    keras.callbacks.ReduceLROnPlateau(
+        monitor="loss", factor=0.5, patience=50, min_lr=0.0001
+    ),
+    keras.callbacks.EarlyStopping(monitor="val_accuracy", patience=50, verbose=1),
+]
+model.compile(
+    optimizer="adam", loss="sparse_categorical_crossentropy", metrics=["accuracy"]
+)
+history = model.fit(
+    x_train,
+    y_train,
+    batch_size=batch_size,
+    epochs=epochs,
+    callbacks=callbacks,
+    validation_split=0.2,
+    verbose=1,
+)
+
+"""
+## Evaluate model on test data
+"""
+
+model = keras.models.load_model("best_model.h5")
+
+y_pred = model.predict(x_test).argmax(axis=1)
+
+print("Test accuracy", accuracy_score(y_pred, y_test))
+
+"""
+## Plot the model's training and validation loss
+"""
+
+metric = "accuracy"
+plt.figure()
+plt.plot(history.history[metric])
+plt.plot(history.history["val_" + metric])
+plt.title("model " + metric)
+plt.ylabel(metric, fontsize="large")
+plt.xlabel("epoch", fontsize="large")
+plt.legend(["train", "val"], loc="best")
+plt.show()
+plt.close()
+
+"""
+We can see how the training accuracy increases and reaches almost 0.9 after 10 epochs. 
+Nevertheless, by observing the validation accuracy we can see how the network still needs
+training until it reaches almost 0.9 for both the validation and the training accuracy 
```

Comment:
This suggests that the network hasn't started overfitting. After how many epochs do you start seeing overfitting?

Reply:
I have removed early stopping to observe the overfitting phenomena (with a batch size of 32 as you suggested). 
It seems like after the 200th epoch, the validation accuracy stops improving while the training accuracy continues on increasing. 
This would suggest that overfitting starts at epoch 200. 
I updated the description accordingly. 
You can observe the validation and training accuracy below:
![image](https://user-images.githubusercontent.com/29229602/91436287-184a8080-e868-11ea-8604-e34708275883.png)